### PR TITLE
Update README and add license headers

### PR DESCRIPTION
**Pull request by mathialo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,113 @@
...
+            metadata=event.metadata,
+            source=event.source,
+        )
+
+with extractor:
+    extractor.run()
+
+```
+
+A full example is provided in th `example.py` file.
```
Comment (emelvarol-cognite) [inline README.md:79]:
```suggestion
A full example is provided in the `example.py` file.
```

maybe link example.py file here?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,113 @@
 # Cognite `extractor-utils` REST extension
```
Comment (emelvarol-cognite) [inline README.md:1]: is this published in pyPI, if so can you add an installation step here?

File: README.md [hunk 3]
```diff
@@ -1,4 +1,113 @@
...
+
+
+## Overview
+
+The REST extension for extractor utils templetizes how the extractor will make HTTP requests to the source,
+automatically serializes the response into user-defined DTO classes, and handles uploading of data to CDF.
+
+The only part of the extractor necessary to for a user to implement are
+
+ * Describing how HTTP requests should be constructed using pre-built function decorators
```
Comment (emelvarol-cognite) [inline README.md:16]:
```suggestion
 * Describing how HTTP requests should be constructed using pre-built `function decorators`
```
hahaha :D

**Review by vvemel — APPROVED** [review]
nicee

File: README.md (hunk 2 above)
Comment (mathialo) [reply README.md:1]: Not yet :) Will add it when we do!

**Pull request merged**